components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/java/org/wso2/carbon/identity/api/server/application/management/v1/core/functions/application/ServiceProviderToApiModel.java

```diff
@@ -439,4 +442,36 @@ private ApplicationResponseModel.AccessEnum getAccess(String applicationName) {
 
         return ApplicationResponseModel.AccessEnum.WRITE;
     }
+
+    private String getClientId(ServiceProvider application) {
+
+        if (application.getInboundAuthenticationConfig() != null) {
+            InboundAuthenticationRequestConfig[] authRequestConfigs = application.getInboundAuthenticationConfig()
+                    .getInboundAuthenticationRequestConfigs();
+
+            if (authRequestConfigs != null && authRequestConfigs.length > 0) {
+                if (authRequestConfigs[0].getInboundAuthType().equals("oauth2")) {
+                    return authRequestConfigs[0].getInboundAuthKey();
+                }
+            }
+        }
+
+        return StringUtils.EMPTY;
+    }
+
+    private String getIssuer(ServiceProvider application) {
+
+        if (application.getInboundAuthenticationConfig() != null) {
+            InboundAuthenticationRequestConfig[] authRequestConfigs = application.getInboundAuthenticationConfig()
+                    .getInboundAuthenticationRequestConfigs();
+
+            if (authRequestConfigs != null && authRequestConfigs.length > 0) {
+                if (authRequestConfigs[0].getInboundAuthType().equals("samlsso")) {
+                    return authRequestConfigs[0].getInboundAuthKey();
+                }
+            }
+        }
```

Comment:
code line 448-459 and 464-475 seems to be duplicated. We can write a common method, passing the auth type(`samlsso`, `oauth2`) as a variable

Reply:
Addressed this via : https://github.com/wso2/identity-api-server/pull/418